# Applies C++ name mangling to all of OpenAL Soft's functions so that they don't conflict with the functions in Emscripten's OpenAL implementation.
# Otherwise, the Emscripten build crashes on startup because the Emscripten version of RetroArch also depends on OpenAL, and it will incorrectly try to use our OpenAL implementation instead of Emscripten's.

--- a/include/AL/al.h
+++ b/include/AL/al.h
@@ -2,7 +2,7 @@
 #define AL_AL_H
 
 #ifdef __cplusplus
-extern "C" {
+
 #endif
 
 #ifndef AL_API
@@ -668,7 +668,7 @@ typedef void          (AL_APIENTRY *LPALSPEEDOFSOUND)(ALfloat value);
 typedef void          (AL_APIENTRY *LPALDISTANCEMODEL)(ALenum distanceModel);
 
 #ifdef __cplusplus
-}  /* extern "C" */
+
 #endif
 
 #endif /* AL_AL_H */
--- a/include/AL/alc.h
+++ b/include/AL/alc.h
@@ -2,7 +2,7 @@
 #define AL_ALC_H
 
 #ifdef __cplusplus
-extern "C" {
+
 #endif
 
 #ifndef ALC_API
@@ -268,7 +268,7 @@ typedef void           (ALC_APIENTRY *LPALCCAPTURESTOP)(ALCdevice *device);
 typedef void           (ALC_APIENTRY *LPALCCAPTURESAMPLES)(ALCdevice *device, ALCvoid *buffer, ALCsizei samples);
 
 #ifdef __cplusplus
-} /* extern "C" */
+
 #endif
 
 #endif /* AL_ALC_H */
--- a/include/AL/alext.h
+++ b/include/AL/alext.h
@@ -22,7 +22,7 @@ typedef uint64_t _alsoft_uint64_t;
 #include "al.h"
 
 #ifdef __cplusplus
-extern "C" {
+
 #endif
 
 #ifndef AL_LOKI_IMA_ADPCM_format
@@ -649,7 +649,7 @@ void AL_APIENTRY alSourcePlayAtTimevSOFT(ALsizei n, const ALuint *sources, ALint
 #endif
 
 #ifdef __cplusplus
-}
+
 #endif
 
 #endif
--- a/include/AL/efx.h
+++ b/include/AL/efx.h
@@ -7,7 +7,7 @@
 #include "al.h"
 
 #ifdef __cplusplus
-extern "C" {
+
 #endif
 
 #define ALC_EXT_EFX_NAME                         "ALC_EXT_EFX"
@@ -756,7 +756,7 @@ AL_API void AL_APIENTRY alGetAuxiliaryEffectSlotfv(ALuint effectslot, ALenum par
 
 
 #ifdef __cplusplus
-}  /* extern "C" */
+
 #endif
 
 #endif /* AL_EFX_H */
